dev/import_beats/main.go

```diff
@@ -0,0 +1,219 @@
+	flag.Parse()
+
+	if beatsDir == "" || publicDir == "" {
+		log.Fatal("beatsDir and publicDir must be set")
+	}
+
+	integrations := NewIntegrations()
+	for beat, category := range BEAT_CATEGORIES {
+		dirs, err := ioutil.ReadDir(filepath.Join(beatsDir, beat, "module"))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		for _, module := range dirs {
+			if !module.IsDir() {
+				continue
+			}
+
+			if _, ok := IGNORE[module.Name()]; ok {
+				continue
+			}
+
+			integrations.AddBeatModule(category, filepath.Join(beatsDir, beat, "module", module.Name()))
+		}
+	}
+
+	integrations.Write(publicDir)
+}
+
+type Integration struct {
+	util.Package
+	Path string
+}
+
+type Integrations struct {
+	list map[string]*Integration
+}
+
+// NewIntegrations returns an new empty list of integrations
+func NewIntegrations() *Integrations {
+	return &Integrations{
+		list: make(map[string]*Integration, 0),
+	}
+}
+
+// AddBeatModule reads info from the module folder into the integrations structure
+func (i *Integrations) AddBeatModule(category, path string) {
+	meta := readMeta(path)
+	if meta == nil {
+		return
+	}
+
+	integration, ok := i.list[meta.Key]
+	if !ok {
+		integration = &Integration{
+			Package: util.Package{
+				Name: meta.Key,
+				// TODO use stack version?
+				Version: "1.0.0",
+				Title:   &meta.Title,
+				Requirement: util.Requirement{
+					Kibana: util.Kibana{
+						Version: util.Version{
+							Min: "6.7.0",
+							// TODO do we really require a max version?
+							Max: "7.6.0",
+						},
+					},
+				},
+			},
+			Path: path,
+		}
+		i.list[meta.Key] = integration
+	}
+
+	integration.Categories = append(integration.Categories, category)
+
+	// TODO come up with a general description
+	integration.Description = meta.Description
+}
+
+// Write integration files & manifests to the given destination folder
+func (i *Integrations) Write(destination string) {
+	log.Println("Writing integration manifests and files to " + destination)
+
+	for _, integration := range i.list {
+		path := filepath.Join(destination, integration.Name+"-"+integration.Version)
+		err := os.MkdirAll(path, 0755)
+		if err != nil {
+			log.Fatal("Could not create integration folder: ", err)
+		}
+
+		// Write manifest.yml
+		data, err := yaml.Marshal(integration.Package)
+		if err != nil {
+			log.Fatal("Could not marshal integration manifest.yaml: ", err)
+		}
+
+		err = ioutil.WriteFile(filepath.Join(path, "manifest.yml"), data, 0644)
+		if err != nil {
+			log.Fatal("Could not write integration manifest.yaml: ", err)
+		}
+
+		// Copy assets
+
+		// dashboards
+		srcDashboardPath := filepath.Join(integration.Path, "_meta/kibana/7/dashboard")
+		fmt.Println(srcDashboardPath)
+		srcDashboards, err := ioutil.ReadDir(srcDashboardPath)
+		if err != nil && !os.IsNotExist(err) {
+			log.Fatal(err)
+		}
+
+		if len(srcDashboards) > 0 {
+			dstDashboardPath := filepath.Join(path, "kibana", "dashboard")
```

Comment:
We should discuss what is best for you as the desitination dashboard / visualization etc. format. Also should it be encoded or not?

Reply:
That's a good question, so far it has been good for development and keeping things in a repo. I would be +1 on keeping them as a pretty json

Reply:
I definitively think we should keep it as pretty json for the development of integrations. What about inside the packages?

Reply:
I guess as part of the publish process we can do any transformation that is needed. This probably depends on what's best for Kibana?

Reply:
I would argue Kibana wants to do as little work as possible so getting already the "right" format would be beneficial form them I would argue.

One thing to take into account here is Kibana will potentially move to the decoded format in the future ...

Reply:
My thinking at the moment:

* Decoded in the integrations repo where you build it
* Decoded in the unzippped packages we have here at the moment
* Encoded inside to public dir.